packages/admin/cms-admin/src/common/MasterMenu.tsx

```diff
@@ -91,25 +148,60 @@ export const MasterMenu: React.FC<MasterMenuProps> = ({ menu, permanentMenuMinWi
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [location]);
 
+    const renderSubmenuItems = (submenu: MenuCollapsibleElement["submenu"]) =>
+        submenu.flatMap((submenuItem, index) => {
+            if (isMenuItemCollapsible(submenuItem)) {
+                return (
+                    <MenuCollapsibleItem key={index} {...submenuItem.menuElement}>
+                        {renderSubmenuItems(submenuItem.submenu)}
+                    </MenuCollapsibleItem>
+                );
+            } else if (isMenuItemAnchor(submenuItem)) {
+                return <MenuItemAnchorLink key={index} {...submenuItem.menuElement} />;
+            } else if (isMenuItemRoute(submenuItem)) {
+                return <MenuItemRouterLink key={index} {...submenuItem.menuElement} to={`${match.url}${submenuItem.menuElement.to}`} />;
+            }
+            return [];
+        });
+
+    const renderGroupItems = (groupItems: MenuItemGroupElement["groupItems"]) =>
+        groupItems.flatMap((groupItem, index) => {
+            if (isMenuItemCollapsible(groupItem) && !!groupItem.submenu.length) {
+                return (
+                    <MenuCollapsibleItem key={index} {...groupItem.menuElement}>
+                        {renderSubmenuItems(groupItem.submenu)}
+                    </MenuCollapsibleItem>
+                );
+            } else if (isMenuItemRoute(groupItem)) {
+                return <MenuItemRouterLink key={index} {...groupItem.menuElement} to={`${match.url}${groupItem.menuElement.to}`} />;
+            } else if (isMenuItemAnchor(groupItem)) {
+                return <MenuItemAnchorLink key={index} {...groupItem.menuElement} />;
+            }
+            return [];
+        });
```

Comment:
This two functions do the same, right?

Reply:
Merged in c5e6187df2a8b00633a19c77cf2e35cc69661950